mono/utils/mono-dl.c

```diff
@@ -177,7 +181,11 @@ mono_dl_open (const char *name, int flags, char **error_msg)
 	}
 	module->main_module = name == NULL? TRUE: FALSE;
 
+#if defined(PLATFORM_UNITY)
```

Comment:
All these #if checks are okay, but maybe we should put single #if check at top and extract differences? i.e. define a new API on top of all of these that is consistent across mono and unity platform APIs?